src/dbnode/storage/types.go

```diff
@@ -476,7 +493,8 @@ type namespaceIndex interface {
 		ctx context.Context,
 		query index.Query,
 		opts index.QueryOptions,
-	) (index.QueryResults, error)
+		aggResultOpts *index.AggregateResultsOptions,
```

Comment:
Instead of making this an optional arg, can we just create a second `AggregateQuery(...)` method on `namespaceIndex` too?